remix-solidity/src/compiler/compiler.ts

```diff
@@ -0,0 +1,392 @@
+  }
+
+  onInternalCompilerLoaded () {
+    if (this.state.worker === null) {
+      let compiler
+      if (typeof (window) === 'undefined') {
+        compiler = require('solc')
+      } else {
+        compiler = solc(window['Module'])
+      }
+
+      this.state.compileJSON = (source: SourceWithTarget) => {
+        let missingInputs: any = []
+        let missingInputsCallback = (path) => {
+          missingInputs.push(path)
+          return { error: 'Deferred import' }
+        }
+
+        let result
+        try {
+          if(source && source.sources) {
+            let input = compilerInput(source.sources, {optimize: this.state.optimize, evmVersion: this.state.evmVersion, language: this.state.language})
+            result = compiler.compile(input, { import: missingInputsCallback })
+            result = JSON.parse(result)
+          }
+        } catch (exception) {
+          result = { error: { formattedMessage: 'Uncaught JavaScript exception:\n' + exception, severity: 'error', mode: 'panic' } }
+        }
+        this.compilationFinished(result, missingInputs, source)
+      }
+      this.onCompilerLoaded(compiler.version())
+    }
+  }
+ 
+  /**
+    * return the contract obj of the given @arg name. Uses last compilation result.
+    * return null if not found
+    * @param {String} name    - contract name
+    * @returns contract obj and associated file: { contract, file } or null
+    */
+  getContract (name: string) {
+    if (this.state.lastCompilationResult && this.state.lastCompilationResult.data && this.state.lastCompilationResult.data.contracts) {
+      return txHelper.getContract(name, this.state.lastCompilationResult.data.contracts)
+    }
+    return null
+  }
+
+  /**
+    * call the given @arg cb (function) for all the contracts. Uses last compilation result
+    * @param {Function} cb    - callback
+    */
+  visitContracts (cb: any) {
+    if (this.state.lastCompilationResult && this.state.lastCompilationResult.data && this.state.lastCompilationResult.data.contracts) {
+      return txHelper.visitContracts(this.state.lastCompilationResult.data.contracts, cb)
+    }
+    return null
+  }
+
+  /**
+    * return the compiled contracts from the last compilation result
+    * @return {Object}     - contracts
+    */
+  getContracts () {
+    if (this.state.lastCompilationResult && this.state.lastCompilationResult.data && this.state.lastCompilationResult.data.contracts) {
+      return this.state.lastCompilationResult.data.contracts
+    }
+    return null
+  }
+
+   /**
+    * return the sources from the last compilation result
+    * @param {Object} cb    - map of sources
+    */
+  getSources (){
+    if (this.state.lastCompilationResult && this.state.lastCompilationResult.source) {
+      return this.state.lastCompilationResult.source.sources
+    }
+    return null
+  }
+
+  /**
+    * return the sources @arg fileName from the last compilation result
+    * @param {Object} cb    - map of sources
```

Comment:
Update the comment. It doesn't match the type.
Also You don't need to specify the type `{ Object }` for tsdoc

Reply:
done